src/main/java/software/amazon/smithy/lsp/ServerArguments.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.lsp;
+
+import java.util.function.Consumer;
+import software.amazon.smithy.cli.AnsiColorFormatter;
+import software.amazon.smithy.cli.ArgumentReceiver;
+import software.amazon.smithy.cli.Arguments;
+import software.amazon.smithy.cli.CliError;
+import software.amazon.smithy.cli.CliPrinter;
+import software.amazon.smithy.cli.HelpPrinter;
+
+/**
+ * Options and Params available for LSP.
+ */
+final class ServerArguments implements ArgumentReceiver {
+
+    static final int MIN_PORT = 0;
+    static final int MAX_PORT = 65535;
+    static final int DEFAULT_PORT = 0; // Default value for unset port number.
+    static final String HELP = "--help";
+    static final String HELP_SHORT = "-h";
+    static final String PORT_NUMBER = "--port-number";
+    static final String PORT_NUMBER_SHORT = "-p";
+    static final String PORT_NUMBER_POSITIONAL = "<port_number>";
```

Comment:
I think with my other suggestion below to add a `useSocket()` method, these can all be made private